### PR TITLE
build(improvements): vue-eslint + setup of vue tests/eslint + exclude core from eslint and tests

### PR DESCRIPTION
**Pull request by Shyrro**
- Added vue-eslint
- setup of tests with vue + config of jest/vue-test-utils 
- Configured eslint/tests to ignore core folder

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
   "extends": [
     "plugin:@typescript-eslint/eslint-recommended",
     "plugin:@typescript-eslint/recommended",
-    "plugin:import/typescript",
-    "prettier"
+    "plugin:jest/recommended",
+    "plugin:import/typescript",    
+    "prettier",
+    "eslint:recommended",
+    "plugin:vue/recommended"
```
Comment (miguelramos) [inline .eslintrc:10]: @Shyrro can you check this: https://eslint.vuejs.org/user-guide/#usage. If it is for vue 3 i think we should use what docs says.

Comment (Shyrro) [reply .eslintrc:10]: Oh great, i didn't know there was a specific version for vue3, thanks ! I'll do that and push a new version of the code !

**Comment by miguelramos** [comment]
@Shyrro look great. Can you rebase locally to resolve conflicts?

**Comment by Shyrro** [comment]
@miguelramos Will do tonight !

**Pull request merged**